High peripheral mode

hvenev opened the pull request:
The first patch just changes the `#size-cells` of `&scb`. The second patch is the one that changes addresses. Both seem to work (ethernet + usb + no errors in dmesg)

I haven't tested it, but I suspect it may be possible to do the second patch as an overlay instead.

hvenev committed:
dts: bcm2711: set #size-cells = <2>

There already is one 4 GiB range, and one more will appear when high
peripheral mode is enabled.

Signed-off-by: Hristo Venev <hristo@venev.name>

hvenev commented on arch/arm/boot/dts/bcm2711-rpi.dtsi:
```diff
@@ -185,17 +185,15 @@
+	ranges = <0x0 0x7c000000  0x4 0x7c000000  0x0 0x04000000>,
+		 <0x0 0x40000000  0x4 0xc0000000  0x0 0x00800000>,
+		 <0x6 0x00000000  0x6 0x00000000  0x0 0x40000000>,
+		 <0x0 0x00000000  0x0 0x00000000  0x0 0x80000000>;
```
Oops. I'll fix the size here.

hvenev replied:
Actually, what is this supposed to mean? There seems to be overlap in the bus addresses in the original DTS. Should the SDRAM ranges even be here?

pelwell replied:
Which bit are you questioning? I can't see a problem.

hvenev replied:
The one that maps 0x0 -> 0x0. I accidentally changed the size from 0xfc000000 to 0x80000000 (instead of 0x100000000).

It looks like that range overlaps with the first two. Should it even be there? None of the nodes inside mention them, and none of the other nodes (except memory) define those ranges.

pelwell replied:
It's probably unnecessary - delete it.

hvenev committed:
dts: bcm2711: remove unused memory range

Physical addresses in the 0x0-0xfc00_0000 range correspond to SDRAM. They
have no place here.

Signed-off-by: Hristo Venev <hristo@venev.name>

hvenev committed:
dts: bcm2711: add "High Peripheral" mode overlay

The following addresses change:

 - 0xfc00_0000 -> 0x4_7c00_0000
 - 0xff80_0000 -> 0x4_c000_0000

The range 0xfc00_0000-0xffff_ffff becomes available as system RAM on
devices with >= 4 GiB of RAM. Firmware should initialize the memory node
appropriately.

Signed-off-by: Hristo Venev <hristo@venev.name>

hvenev commented:
I added a commit that removes the unnecessary range, but I've only tested high peripheral mode.

hvenev commented:
The overlay compiles with warnings about missing `#size-cells` and `#address-cells`. Those need to be specified both in the node and its parent to silence the warning.

pelwell commented:
I was going to apologise about asking for the old syntax because my checker tools don't understand the new format yet, but in this case the old syntax allows you to get rid of the warnings. I've updated the PR with an equivalent overlay in the old style. If you approve, I'll merge the PR.

hvenev commented:
LGTM

pelwell commented:
Thanks again.
